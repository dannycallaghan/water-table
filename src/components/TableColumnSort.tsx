import { ReactNode, memo, useRef } from 'react';
import { ISortOptions } from '../types/types';

const TableColumnSort = memo(function TableColumnSort(props: {
  children: ReactNode;
  values?: Omit<ISortOptions, 'desc'>;
  active?: boolean;
  action: (values: ISortOptions) => void;
}) {
  const { action } = props;
  const desc = useRef<boolean>(true);

  const handleSort = (): void => {
    desc.current = !desc.current;
    const values = {
      ...props.values,
      desc: desc.current,
    };
    action(values as never);
  };

  return (
    <div
      className="flex cursor-pointer flex-row items-center gap-1"
      onClick={handleSort}
    >
      {props.children}
      <div className={`${props.active ? 'visible' : 'invisible'}`}>
        {desc.current ? (
          <svg
            xmlns="http://www.w3.org/2000/svg"
            fill="none"
            viewBox="0 0 24 24"
            strokeWidth={1.5}
            stroke="currentColor"
            className="size-5"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              d="m19.5 8.25-7.5 7.5-7.5-7.5"
            />
          </svg>
        ) : (
          <svg
            xmlns="http://www.w3.org/2000/svg"
            fill="none"
            viewBox="0 0 24 24"
            strokeWidth={1.5}
            stroke="currentColor"
            className="size-5"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              d="m4.5 15.75 7.5-7.5 7.5 7.5"
            />
          </svg>
        )}
      </div>
    </div>
  );
});

export default TableColumnSort;
