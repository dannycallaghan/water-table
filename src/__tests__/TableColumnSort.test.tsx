import { expect, test, describe } from 'vitest';
import { render } from '@testing-library/react';
import TableColumnSort from '../components/TableColumnSort.tsx';

describe('TableColumnSort', () => {
  const sortAction = () => {};
  const sortValues = { type: 'number', key: 'bytes' };
  test('renders correctly', () => {
    const sortActive = true;
    const { asFragment } = render(
      <TableColumnSort
        action={sortAction}
        values={sortValues}
        active={sortActive}
      >
        <div></div>
      </TableColumnSort>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders correctly', () => {
    const sortActive = false;
    const { asFragment } = render(
      <TableColumnSort
        action={sortAction}
        values={sortValues}
        active={sortActive}
      >
        <div></div>
      </TableColumnSort>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
