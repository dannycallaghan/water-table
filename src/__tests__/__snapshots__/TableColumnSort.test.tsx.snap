// Vitest Snapshot v1

exports[`TableColumnSort > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex cursor-pointer flex-row items-center gap-1"
  >
    <div />
    <div
      class="visible"
    >
      <svg
        class="size-5"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m19.5 8.25-7.5 7.5-7.5-7.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TableColumnSort > renders correctly 2`] = `
<DocumentFragment>
  <div
    class="flex cursor-pointer flex-row items-center gap-1"
  >
    <div />
    <div
      class="invisible"
    >
      <svg
        class="size-5"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m19.5 8.25-7.5 7.5-7.5-7.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
